docs/static/persistent-queues.asciidoc

```diff
@@ -1,44 +1,228 @@
+
+`queue.checkpoint.acks`:: Sets the number of acked events before forcing a checkpoint. 
+Default is `1024`. Set to `0` for unlimited.
+`queue.checkpoint.writes`:: Sets the maximum number of written events before a forced checkpoint. 
+Default is `1024`. Set to `0` for unlimited.
++
+To avoid losing data in the persistent queue, you can set `queue.checkpoint.writes: 1` to force a checkpoint after each event is
+written. Keep in mind that disk writes have a resource cost. Setting this value
+to `1` ensures maximum durability, but can severely impact performance.
+See <<durability-persistent-queues>> to better understand the trade-offs.
+`queue.checkpoint.interval`:: Sets the interval in milliseconds when a checkpoint is forced on the head page.
+Default is `1000`. Set to `0` to eliminate periodic checkpoints.
+
+[[pq-config-notes]]
+==== Configuration notes
+
+Every situation and environment is different, and the "ideal" configuration varies.
+If you optimize for performance, you may increase your risk of losing data.
+If you optimize for data protection, you may impact performance. 
+
+[[pq-size]]
+===== Queue size
+You can control queue size with the `queue.max_events` and  `queue.max_bytes` settings.
+If both settings are specified, Logstash uses whichever criteria is reached
+first. 
+See <<backpressure-persistent-queue>> for behavior when queue limits are
+reached.
+
+[[pq-lower-max_bytes]]
+===== Smaller queue size
+If you are using persistent queues to protect against data loss, but don't
+require much buffering, you can set `queue.max_bytes` to a smaller value, such
+as 10mb.
+A smaller value produces smaller queues and improves queue performance.
+
+*Sample configuration*
+
+[source, yaml]
+-----
+queue.type: persisted
+queue.max_bytes: 10mb 
+-----
+
+[[pq-fewer-checkpoints]]
+===== Fewer checkpoints
+
+Setting `queue.checkpoint.writes` and `queue.checkpoint.acks` to `0` may
+yield maximum performance, but may have potential impact on durability.
+
+In a situation where Logstash is terminated or there is a hardware-level
+failure, any data that has not been checkpointed, is lost. 
+See <<durability-persistent-queues>> to better understand the trade-offs.
+
+[[troubleshooting-pqs]]
+==== Troubleshooting persistent queues
+
+If you see an error message similar to this one, you likely have a persistent queue problem.
+
+```
+message=>"java.io.IOException: Page file size is too small to hold elements
+```
+
+This error indicates that the head page (the oldest in a folder and the one with lowest page id) has a size < 18 bytes, the size of a page header.
+
+
+To resolve the issue: 
+
+. Identify the queue (or queues) that may be corrupt.
+. Stop Logstash.
+. Run `pqrepair <path>` for each of the corrupted queues.
+
+[[pqcheck]]
+===== `pqcheck` utility
 
-The following are problems not solved by the persistent queue feature:
+Run the `pqcheck` utility to identify which persistent queue--or queues--have been corrupted. 
```

Comment:
Shouldn't we explain a bit more what pqcheck does and how it does it? e.g.:

The `pqcheck` check utility allows users to identify which persistent queue(s) have been corrupted. This tool checks [...] by doing [...] or comparing [...].

Generally speaking, simply improving the description of what this tool does.